CountOnMeTests/CalculatorModelTests.swift

```diff
@@ -0,0 +1,106 @@
+
+final class CalculatorManagerTests: XCTestCase {
+    var calculatorModel: CalculatorModel!
+    let numberFormatter: NumberFormatter = .init()
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+        self.calculatorModel = .init()
+        numberFormatter.locale = .current
+        numberFormatter.decimalSeparator = "."
+        numberFormatter.minimumFractionDigits = 4
+    }
+
+    func testStringSeparator() throws {
+        let rawString: String = "2 - 1 + 6 x 10 - 4 / 2 = "
+        let actualResult = calculatorModel.testSeparateStrings(rawString: rawString)
+        let expectedResult: [Substring] = ["2", "-", "1", "+", "6", "x", "10", "-", "4", "/", "2", "="]
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testRemovePrioritiesFromString() throws {
+        let rawString: String = "2 - 1 + 6 x 10 - 4 / 2 = "
+        calculatorModel.testApplyPriorityRules(rawString: rawString)
+        let expectedResult: [Substring] = ["2", "-", "1", "+", "60.0", "-", "2.0", "="]
+
+        XCTAssertEqual(calculatorModel.getStringResult(), expectedResult)
+    }
+
+    func testAddStringOperands() {
+        let leftOperand: Substring = .init("12")
+        let rightOperand: Substring = .init("2")
+        let actualResult = Operations.add(leftOperand, rightOperand)
+        let expectedResult: Substring = .init("14.0")
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testSubtractStringOperands() {
+        let leftOperand: Substring = .init("12")
+        let rightOperand: Substring = .init("2")
+        let actualResult = Operations.substract(leftOperand, rightOperand)
+        let expectedResult: Substring = .init("10.0")
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testMultiplyStringOperands() {
+        let leftOperand: Substring = .init("12")
+        let rightOperand: Substring = .init("2")
+        let actualResult = Operations.multiply(leftOperand, rightOperand)
+        let expectedResult: Substring = .init("24.0")
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testDivideStringOperands() {
+        let leftOperand: Substring = .init("12")
+        let rightOperand: Substring = .init("2")
+        let actualResult = Operations.divide(leftOperand, rightOperand)
+        let expectedResult: Substring = .init("6.0")
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testGetResultFromCalculator() {
+        let rawString: String = "2 x 3 + 1 - 10 / 5 = "
+        let expectedResult: String = "5.0"
+        let actualResult = calculatorModel.getResult(rawString: rawString)
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testGetResultFromCalculatorWithFormattedDecimalNumber() {
+        let rawString: String = "4 / 3 - 1 = "
+        let expectedResult: String = "0.3333"
```

Comment:
**nitpick (non-blocking)**: Save yourself some time and some headaches, remove the unnecessary types:

```suggestion
        let rawString = "4 / 3 - 1 = "
        let expectedResult = "0.3333"
```

Reply:
This is an App wide suggestion 😉